egs/cnceleb/v2/run.sh

```diff
@@ -0,0 +1,151 @@
+
+  # Now we're ready to create training examples.
+  utils/fix_data_dir.sh data/train_comb_no_sil
+fi
+  
+# Stages 6 through 8 are handled in run_xvector.sh
+local/nnet3/xvector/run_xvector.sh --stage $stage --train-stage -1 \
+  --data data/train_comb_no_sil --nnet-dir $nnet_dir \
+  --egs-dir $nnet_dir/egs
+
+if [ $stage -le 9 ]; then
+  # These x-vectors will be used for mean-subtraction, LDA, and PLDA training.
+  sid/nnet3/xvector/extract_xvectors.sh --cmd "$train_cmd --mem 4G" --nj 20 \
+    $nnet_dir data/train_comb \
+    $nnet_dir/xvectors_train_comb
+
+  # Extract x-vector for eval sets.
+  for name in eval_enroll eval_test; do
+    sid/nnet3/xvector/extract_xvectors.sh --cmd "$train_cmd --mem 4G" --nj 20 \
+      $nnet_dir data/$name \
+      $nnet_dir/xvectors_$name
+  done
+fi
+
+if [ $stage -le 10 ]; then
+  # Compute the mean.vec used for centering.
+  $train_cmd $nnet_dir/xvectors_train_comb/log/compute_mean.log \
+    ivector-mean scp:$nnet_dir/xvectors_train_comb/xvector.scp \
+    $nnet_dir/xvectors_train_comb/mean.vec || exit 1;
+
+  # Use LDA to decrease the dimensionality prior to PLDA.
+  lda_dim=128
+  $train_cmd $nnet_dir/xvectors_train_comb/log/lda.log \
+    ivector-compute-lda --total-covariance-factor=0.0 --dim=$lda_dim \
+    "ark:ivector-subtract-global-mean scp:$nnet_dir/xvectors_train_comb/xvector.scp ark:- |" \
+    ark:data/train_comb/utt2spk $nnet_dir/xvectors_train_comb/transform.mat || exit 1;
+
+  # Train the PLDA model.
+  $train_cmd $nnet_dir/xvectors_train_comb/log/plda.log \
+    ivector-compute-plda ark:data/train_comb/spk2utt \
+    "ark:ivector-subtract-global-mean scp:$nnet_dir/xvectors_train_comb/xvector.scp ark:- | transform-vec $nnet_dir/xvectors_train_comb/transform.mat ark:- ark:- | ivector-normalize-length ark:- ark:- |" \
+    $nnet_dir/xvectors_train_comb/plda || exit 1;
+fi
+
+if [ $stage -le 11 ]; then
+  # Compute PLDA scores for CN-Celeb eval core trials
+  $train_cmd $nnet_dir/scores/log/cnceleb_eval_scoring.log \
+    ivector-plda-scoring --normalize-length=true \
+    --num-utts=ark:$nnet_dir/xvectors_eval_enroll/num_utts.ark \
+    "ivector-copy-plda --smoothing=0.0 $nnet_dir/xvectors_train_comb/plda - |" \
+    "ark:ivector-mean ark:data/eval_enroll/spk2utt scp:$nnet_dir/xvectors_eval_enroll/xvector.scp ark:- | ivector-subtract-global-mean $nnet_dir/xvectors_train_comb/mean.vec ark:- ark:- | transform-vec $nnet_dir/xvectors_train_comb/transform.mat ark:- ark:- | ivector-normalize-length ark:- ark:- |" \
+    "ark:ivector-subtract-global-mean $nnet_dir/xvectors_train_comb/mean.vec scp:$nnet_dir/xvectors_eval_test/xvector.scp ark:- | transform-vec $nnet_dir/xvectors_train_comb/transform.mat ark:- ark:- | ivector-normalize-length ark:- ark:- |" \
+    "cat '$eval_trials_core' | cut -d\  --fields=1,2 |" $nnet_dir/scores/cnceleb_eval_scores || exit 1;
+
+  # CN-Celeb Eval Core:
+  # EER: 14.70%
+  # minDCF(p-target=0.01): 0.6814
+  # minDCF(p-target=0.001): 0.7979
```

Comment:
It's quite unexpected that a well designed x-vector system would produce worse results than a traditional i-vector system. 

I suggest changing this PR so that it only includes the v1 recipe for now. Postpone including the v2 recipe until you know what's wrong with the x-vector system or until you have a better understanding of how to train these kinds of systems. I haven't had time to look at everything that could be improved in the recipe, but I notice that the PR doesn't even include data augmentation in the x-vector training recipe, which we know is an essential step for achieving good performance.

Reply:
In addition to augmentation, you might wish to include some other free, wideband resource, like VoxCeleb in the DNN training data. My guess is that this will substantially improve the x-vector system performance, even though it introduces a language mismatch. You'll still want to use the in-domain data for training the PLDA model, though.

Alternatively, if you feel the recipe should consist of nothing but the CN-Celeb dev and eval datasets, then I would again consider removing the v2 recipe, but retain the v1 recipe. If you do choose to use only this data, then it could be the case that the v2 recipe is simply not appropriate for it.

Reply:
Agree! I have removed the v2 recipe.

Actually, we have tried our best to improve the performance on the x-vector system if only choosing the CN-Celeb dev for x-vector training. Data augmentation similar as egs/VoxCeleb does not work in CN-Celeb training in our experiment. While in-domain PLDA training with the x-vector trained with VoxCeleb is effective
Related experimental results can be seen [https://arxiv.org/abs/1911.01799](url).

We hope the recipe only consist of the CN-Celeb dev and eval datasets, and accept your suggestion to remove the v2 recipe.

Reply:
Thanks! You could always decide to add an x-vector recipe later... and then I'd suggest doing whatever produces the best performance, which as you said, appears to be training the DNN on VoxCeleb (or perhaps VoxCeleb + CN-Celeb?) and the backend on CN-Celeb.

Reply:
Thanks! I will try it later to make the best performance on x-vector.